modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/TableScanNode.java

```diff
@@ -184,7 +184,13 @@ private void push() throws Exception {
             }
         }
 
+        assert waiting >= 0;
```

Comment:
I don't sure it's a good idea to put assertions randomly in the code. If need to validate some condition, then put validation as close as it possible to the place where the invariant of interest might be put to an invalid state